src/components/App.tsx

```diff
@@ -320,26 +337,28 @@ export class App extends React.Component<any, AppState> {
                       localElementMap[element.id].version > element.version
                     ) {
                       elements.push(localElementMap[element.id]);
+                      delete localElementMap[element.id];
                     } else {
-                      elements.push(element);
+                      if (deletedIds.hasOwnProperty(element.id)) {
+                        if (element.version > deletedIds[element.id].version) {
+                          elements.push(element);
+                          delete deletedIds[element.id];
+                          delete localElementMap[element.id];
+                        }
+                      } else {
+                        elements.push(element);
+                        delete localElementMap[element.id];
+                      }
                     }
 
                     return elements;
                   }, [] as any)
-                  // add local elements that are currently being edited
-                  // (can't be done in the step above because the elements may
-                  //  not exist on remote at all)
-                  .concat(
-                    elements.filter(element => {
-                      return (
-                        element.id === this.state.editingElement?.id ||
-                        element.id === this.state.resizingElement?.id ||
-                        element.id === this.state.draggingElement?.id
-                      );
-                    }),
-                  );
```

Comment:
Why do you not need this anymore? It seems different from the deleted elements?

Reply:
I'm concatting it with the remaining elements from `localElementMap`. That should cover everything from the previous, and more.

Reply:
Ok!